### DISCUSSION TITLE
i3 exec not behaving like I thought it did

### DISCUSSION THREAD

**Discussion by jamesonBradfield** [i3/i3#5892] [Q&A] [answered]
ok, so I might me misunderstanding the help file for nvim but I have this bash script that runs a couple of commands to figure out directory and runs ObsidianNew after the config
```bash
#!/bin/bash
cd /
pwd
echo "folder:(options are School,Personal,Liesure) or (s,p,l)"
read input
if [[ "$input" == "l" ]] || [[ "$input" == "Liesure" ]]; then
  cd /media/jamie/bigChonkie/ObsidianVault/MyVault/Leisure
elif [[ "$input" == "p" ]] || [[ "$input" == "Personal" ]]; then
  cd /media/jamie/bigChonkie/ObsidianVault/MyVault/Personal
elif [[ "$input" == "s" ]] || [[ "$input" == "School" ]]; then
  cd /media/jamie/bigChonkie/ObsidianVault/MyVault/School
else
  echo "you input something wrong. closing bash"
  exit 0
fi

nvim -c ObsidianNew
```
and I'm running it in i3 with `exec i3-sensible-terminal --hold -e /home/jamie/bin/take-note.sh `and I get 
```bash
/home/jamie/bin/take-note.sh: line 17: nvim: command not found
```
I know this might not be the right place to ask this but what the heck is going on?
running the command in my terminal works like I expect.

**Comment by orestisfl** [comment]
Run `command -v nvim` from the terminal. Where is it installed? Is that path added to the `$PATH` environmental variable when i3 is started?

Make sure to set-up any `export PATH=...` commands you might have in your dotfiles in a place where they will take effect before i3 is started. For example, `~/.xinitrc` or `~/.xsession` or `~/.bash_profile` depending on how you start i3 and your shell preferences.

**Reply by jamesonBradfield** [reply]
hi orestifl I used brew to install neovim, and have said environment variables in my path I am not sure how I  would go about setting that up for my xinitrc I have tried the steps here(https://docs.brew.sh/Homebrew-on-Linux) and piping it to xinitrc, instead of bashrc, that didn't seem to work, 
my nvim install is in /home/linuxbrew/.linuxbrew/bin/nvim
thanks again for your help!

**Reply by orestisfl** [reply]
How do you start i3? startx? Some graphical environment?

**Reply by jamesonBradfield** [reply]
just through clicking the Icon in login and logging in.

**Reply by orestisfl** [reply]
There is no such thing in i3, you are using some kind of login manager which I assume is configured by default in your distribution. We need to know what that is to figure out how you can configure the $PATH.

Otherwise, neovim is quite easily available for most distros. You can try to install it through your distro's package manage instead of brew.

**Reply by jamesonBradfield** [reply]
I am using linux mint and whatever it's default login manager is, sudo apt install neovim installs version .6 which is way too outdated for my needs.

**Reply by orestisfl** [reply, marked as answer]
> and whatever it's default login manager is

From what i can tell, linux mint uses lightdm, which should be sourcing `~/.profile` and/or `~/.xprofile`. Try adding `export PATH="$PATH:/home/linuxbrew/.linuxbrew/bin"` in either file.

**Reply by jamesonBradfield** [reply]
that did it!, thanks soo much!
